mainframer.sh

```diff
@@ -70,6 +77,33 @@ if [ -z "$REMOTE_COMMAND" ]; then
 	exit 1
 fi
 
+function convertSecondsToTime {
+((h=${1}/3600))
+((m=(${1}%3600)/60))
+((s=${1}%60))
+
+#Find out the time format and correct string.
+if [ "$h" -eq "1" ]; then
+    HOURS_TAG="hour"
+    HOURS_FORMAT="%01d"
+fi
+if [ "$m" -eq "1" ]; then
+    MINUTES_TAG="minute"
+    MINUTES_FORMAT="%01d"
+fi
+if [ "$s" -eq "1" ]; then
+    SECONDS_TAG="second"
+    SECONDS_FORMAT="%01d"
+fi
+
+ #Check if hour is 0, so it doesn't need to be shown.
```

Comment:
So it always shows minutes? Our builds usually take less than 30 seconds so we see minutes very rarely :)